sympy/combinatorics/fp_groups.py

```diff
@@ -360,6 +361,38 @@ def index(self, H, strategy="relator_based"):
             C = self.coset_enumeration(H, strategy)
             return len(C.table)
 
+    def is_cyclic(self):
+        """
+        Return ``True`` if the group is Cyclic.
+
+        Examples
+        ========
+
+        >>> from sympy.combinatorics.free_groups import free_group
+        >>> from sympy.combinatorics.fp_groups import FpGroup
+        >>> F, x, y = free_group("x, y")
+        >>> f = FpGroup(F, [x*y*x**-1*y**-1, y**5, x**4])
+        >>> f.is_cyclic()
+        True
+        >>> f = FpGroup(F, [x**4, y**2, x*y*x**-1*y])
+        >>> f.is_cyclic()
+        False
+
+        """
+        if len(self.generators) == 1:
+            return True
+        if not self.is_abelian:
+            return False
+        for p in primefactors(self.order()):
```

Comment:
The order should probably be finite.

Reply:
Yes, for the current implementation order is finite, For infinite order we probably have to implement `Abelian Invariants` which I have included in my GSoC proposal!